app/views/facilities/_product_list.html.haml

```diff
@@ -12,7 +15,7 @@
               - if session_user.can_override_restrictions?(product) || product.can_be_used_by?(acting_user)
                 = builder.text_field :quantity, value: 0, class: "product_quantity", index: nil
                 = builder.hidden_field :product_id, value: product.id, index: nil
-          = link_to product.name + (product.is_hidden? ? ' (hidden)' : ''), facility_product_path(current_facility, product)
+          = link_to product.name + (product.is_hidden? ? ' (hidden)' : ''), facility_product_path(product.facility, product)
```

Comment:
On second thought, I think what you had with `current_facility || product.facility` is better, because it lets us avoid all the `includes` we would in the facility-specific context to prevent N+1s.

Specifically on this page: https://github.com/tablexi/nucore-open/blob/master/app/views/facilities/show.html.haml